### PR TITLE
Fix broken ramp up and down formulations

### PR DESCRIPTION
**Pull request by nnhjy**
A workaround fix w.r.t. [proposal 2](https://github.com/spine-tools/SpineOpt.jl/pull/566#issuecomment-1533060728) on the [issue](https://github.com/spine-tools/SpineOpt.jl/pull/566#issuecomment-1532304247) discussed under PR #566. The aim is to fix the broken functionality while minimising changes and users' application efforts as new implementation is pending. A more standardized option can be found [here](https://github.com/spine-tools/SpineOpt.jl/pull/566#issuecomment-1533145163)

## Checklist before merging
- [x] Documentation is up-to-date
- [x] Unit tests have been added/updated accordingly
- [x] Code has been formatted according to SpineOpt's style
- [x] Unit tests pass

### CODE REVIEW

**Comment by nnhjy** [comment]
The documentation of the formulations has currently been removed (#647 ). I will update it as soon as it gets restored.

**Comment by mihlema** [comment]
Hi @nnhjy, thanks for taking care of this. I quickly looked through this and the fix seems to make sense to me. Pending the documentation status, we could merge this.
To summarize what this merge request will enable: If `max_startup/shutdown_ramp` parameters are defined, they will be used as an upperbound for ramping during start/shutdown of a unit. If they are not defined, we assume that the unit can ramp up/down full capacity.

File: test/constraints/constraint_unit.jl [hunk 1]
```diff
@@ -1428,6 +1428,31 @@ function test_constraint_ramp_up()
             observed_con = constraint_object(constraint[con_key...])
             @test _is_constraint_equal(observed_con, expected_con)
         end
+
+        max_startup_ramp = 0.4
```
Comment (mihlema) [inline test/constraints/constraint_unit.jl:1432]: Is max_startup_ramp actually used in this test? Seems like the constraint function actually calls ramp_up_limit. Could we have tests that cover the two cases 1) max_startup_ramp defined/not defined?

Comment (mihlema) [reply test/constraints/constraint_unit.jl:1432]: Actually, I guess you are testing the "is defined" case here, and probably the not-defined case has been added beforehand, so should be good

Comment (nnhjy) [reply test/constraints/constraint_unit.jl:1432]: Thanks for the review! Exactly, this block tests the case where `max_startup_ramp` is defined. And the previous block tests the case where it is not defined with the default value `None`.

**Comment by nnhjy** [comment]
Hi @mihlema the documentation has got returned and renewed accordingly. I hope the formulations for this change is correctly written.

**Pull request merged**